src/OAuth/Credential.php

```diff
@@ -92,4 +92,20 @@ public function getRefreshToken(): string
     {
         return $this->refreshToken;
     }
+
+    /**
+     * @param string $accessToken
+     * @param string $refreshToken
+     *
+     * @return Credential
+     */
+    public function withRefreshedTokens(string $accessToken, string $refreshToken): self
```

Comment:
Actually the confusing thing is more of the presence of "Refreshed" and the confusion with "Refresh token". "withNewTokens" ? :) Remove the "refreshed" ke yword